instrumentation/opencensus-shim/testing/src/test/java/io/opentelemetry/opencensusshim/JavaagentInstrumentationTest.java

```diff
@@ -0,0 +1,337 @@
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class JavaagentInstrumentationTest {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @BeforeEach
+  void setup() {
+    // by default in opencensus, a probability sampling is used which is not 100%;
+    // we specifically set this configuration here to always sample to ensure traces are emitted for
+    // our tests
+    Tracing.getTraceConfig()
+        .updateActiveTraceParams(
+            Tracing.getTraceConfig().getActiveTraceParams().toBuilder()
+                .setSampler(Samplers.alwaysSample())
+                .build());
+  }
+
+  @Test
+  void testInterleavedSpansOcFirst() {
+    io.opencensus.trace.Tracer ocTracer = Tracing.getTracer();
+    Tracer otelTracer = testing.getOpenTelemetry().getTracer("test");
+
+    io.opencensus.trace.Span outerSpan = ocTracer.spanBuilder("outer-span").startSpan();
+    Span midSpan;
+    io.opencensus.trace.Span innerSpan;
+
+    outerSpan.putAttribute("outer", AttributeValue.booleanAttributeValue(true));
+
+    try (io.opencensus.common.Scope outerScope = ocTracer.withSpan(outerSpan)) {
+      midSpan =
+          otelTracer
+              .spanBuilder("mid-span")
+              .setSpanKind(SpanKind.INTERNAL)
+              .setAttribute("middle", true)
+              .startSpan();
+      try (Scope midScope = midSpan.makeCurrent()) {
+        innerSpan = ocTracer.spanBuilder("inner-span").startSpan();
+        innerSpan.putAttribute("inner", AttributeValue.booleanAttributeValue(true));
+
+        // make current and immediately close -- avoid empty try block
+        ocTracer.withSpan(innerSpan).close();
+
+        innerSpan.end();
+      } finally {
+        midSpan.end();
+      }
+    } finally {
+      outerSpan.end();
+    }
+
+    Tracing.getExportComponent().shutdown();
+
+    SpanContext outerContext = SpanConverterProxy.mapSpanContext(outerSpan.getContext());
+
+    // expecting 1 trace with 3 spans
+    testing.waitAndAssertTraces(
+        ta ->
+            // ensure each span's attributes haven't seeped into parents or children
+            ta.hasSpansSatisfyingExactly(
+                // outer span
+                sa ->
+                    sa.hasName("outer-span")
+                        .hasNoParent()
+                        .hasAttribute(AttributeKey.booleanKey("outer"), true)
+                        .hasAttributesSatisfying(
+                            OpenTelemetryAssertions.satisfies(
+                                AttributeKey.booleanKey("inner"), AbstractBooleanAssert::isNull),
+                            OpenTelemetryAssertions.satisfies(
+                                AttributeKey.booleanKey("middle"), AbstractBooleanAssert::isNull)),
+                // middle span
+                sa ->
+                    sa.hasName("mid-span")
+                        .hasParentSpanId(outerContext.getSpanId())
```

Comment:
That's a good point. I followed this path only bc it seemed to be a bit more literal and traceable: specifically referencing the `outerContext` rather than its ordinal position in the span list which (at least for me coming in cold) was a cognitive leap.

Happy to update these references if you think there's benefit to this, though. WDYT?

Reply:
We use this pattern pretty much everywhere else in this repo; it'd be more consistent this way.

Reply:
Gotcha, makes sense. Done!